### PR TITLE
Problem: (CRO-503) Slashing rate is not proportional to number of failing validators

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Implimented proportional slashing as outlined in https://github.com/cosmos/cosmos-sdk/blob/sunny/prop-slashing-adr/docs/architecture/adt-014-proportional-slashing.md

### CODE REVIEW

File: chain-abci/tests/abci_app.rs [hunk 1]
```diff
@@ -1446,7 +1446,10 @@ fn begin_block_should_slash_byzantine_validators() {
     validator_pubkey.data = base64::decode("EIosObgfONUsnWCBGRpFlRFq5lSxjGIChRlVrVWVkcE=").unwrap();
 
     let mut validator_voting_power = BTreeMap::new();
-    validator_voting_power.insert(staking_account_address, TendermintVotePower::zero());
+    validator_voting_power.insert(
```
Comment (tomtau) [inline chain-abci/tests/abci_app.rs]: seems unused in tests

Comment (devashishdxt) [reply chain-abci/tests/abci_app.rs]: Removed

**Comment by devashishdxt** [comment]
bors retry

**Pull request merged**